instance/tests/models/test_utils.py

```diff
@@ -486,3 +486,35 @@ class DjangoResourceTest(SimpleResourceTestCase):
     Run the same tests as in SimpleResourceTestCase, but using DjangoResource.
     """
     make_resource = DjangoResource
+
+    def setUp(self):
+        self.make_resource.save = Mock()
+
+    def test_model_field_choices(self):
+        """
+        Test that ModelResourceStateDescriptor produces a sensible set of field choices.
+        """
+        model_field_choices = set(self.make_resource.state.model_field_choices)
+        expected_model_field_choices = set((
+            ('state1', 'State1'),
+            ('state2', 'State2'),
+            ('state3', 'State3'),
+        ))
+        self.assertEqual(model_field_choices, expected_model_field_choices)
```

Comment:
When you rebase the code you might want to tweak this test, since I think Sven has made the result ordered in #53, and we should test for that.